healthcare/v1beta1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceListHistory.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClients;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+
+public class FhirResourceListHistory {
+  private static final String FHIR_NAME = "projects/%s/locations/%s/datasets/%s/fhirStores/%s";
+  private static final JsonFactory JSON_FACTORY = new JacksonFactory();
+  private static final NetHttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+
+  public static void fhirResourceListHistory(String resourceName)
+      throws IOException, URISyntaxException {
+    // String resourceName =
+    //    String.format(
+    //        FHIR_NAME, "project-id", "region-id", "dataset-id", "store-id", "fhir-id");
+
+    // Initialize the client, which will be used to interact with the service.
+    CloudHealthcare client = createClient();
+
+    HttpClient httpClient = HttpClients.createDefault();
+    String uri = String.format(
+        "%sv1beta1/%s/_history", client.getRootUrl(), resourceName);
+    URIBuilder uriBuilder = new URIBuilder(uri)
+        .setParameter("access_token", getAccessToken());
+
+    HttpUriRequest request = RequestBuilder
+        .get()
+        .setUri(uriBuilder.build())
+        .addHeader("Content-Type", "application/fhir+json")
+        .addHeader("Accept-Charset", "utf-8")
+        .addHeader("Accept", "application/fhir+json; charset=utf-8")
+        .build();
+
+    // Execute the request and process the results.
+    HttpResponse response = httpClient.execute(request);
+    HttpEntity responseEntity = response.getEntity();
+    if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+      System.err.print(String.format(
+          "Exception retrieving FHIR history: %s\n", response.getStatusLine().toString()));
+      responseEntity.writeTo(System.err);
+      throw new RuntimeException();
+    }
+    System.out.println("FHIR resource history retrieved: ");
+    responseEntity.writeTo(System.out);
+  }
+
+  private static CloudHealthcare createClient() throws IOException {
+    // Use Application Default Credentials (ADC) to authenticate the requests
+    // For more information see https://cloud.google.com/docs/authentication/production
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault(HTTP_TRANSPORT, JSON_FACTORY)
+            .createScoped(Collections.singleton(CloudHealthcareScopes.CLOUD_PLATFORM));
+
+    // Create a HttpRequestInitializer, which will provide a baseline configuration to all requests.
+    HttpRequestInitializer requestInitializer =
+        request -> {
+          credential.initialize(request);
+          request.setConnectTimeout(60000); // 1 minute connect timeout
+          request.setReadTimeout(60000); // 1 minute read timeout
```

Comment:
1 * 60 * 1000 might be clearer

Reply:
Why do we need to set this?

Reply:
I believe I set this at one point because the calls were hanging for long periods of time, and I needed the tests to fail and not hang indefinitely if the call wasn't being returned.